pkg/rebuild/maven/infer.go

```diff
@@ -168,12 +168,63 @@ func getJarJDK(ctx context.Context, name, version string, mux rebuild.RegistryMu
 		return "", errors.Wrap(err, "reading manifest file")
 	}
 	for _, line := range strings.Split(string(manifestReader), "\n") {
-		if strings.HasPrefix(line, "Build-Jdk:") {
+		if strings.HasPrefix(line, "Build-Jdk:") || strings.HasPrefix(line, "Build-Jdk-Spec:") {
 			_, value, _ := strings.Cut(line, ":")
 			return strings.TrimSpace(value), nil
 		}
 	}
-	return "", nil
+	return approximateJarJDK(zipReader)
+}
+
+func approximateJarJDK(jarZip *zip.Reader) (string, error) {
+	// This is a fallback if the JDK version cannot be determined from the JAR manifest.
+	// It looks for a .class file in the JAR to analyze the bytecode version.
+
+	// Find a .class file in the zip
+	for _, file := range jarZip.File {
+		if strings.HasSuffix(file.Name, ".class") && !strings.Contains(file.Name, "$") {
+			// Found a class file, read its bytes and extract major version
+			classFile, err := file.Open()
+			if err != nil {
+				continue
+			}
+			defer classFile.Close()
+
+			classBytes, err := io.ReadAll(classFile)
+			if err != nil {
+				continue
+			}
+
+			majorVersion, err := getClassFileMajorVersion(classBytes)
+			if err != nil {
+				continue
```

Comment:
we might consider logging this case since it feels exceptional enough that we'd like some evidence that it failed. i suppose we could log the others, as well, but if we were to pick one, my vote would be this.